### DISCUSSION TITLE
Use coinbrew to fetch and extract netlib data?

### DISCUSSION THREAD

**Discussion by jhmgoossens** [coin-or/coinbrew#81] [Q&A] [answered]
I'm trying to add unit testing to the MS Visual Studio solution (windows-ci.yml for msvs) for Clp and other projects, but I'm struggling with the netlib and miplib data.

For these actions, the [fetching is done using coinbrew](https://github.com/coin-or/Clp/blob/54b5d0b0e08dc9ec9aa84c3713f13a47f1a46c15/.github/workflows/windows-ci.yml#L67), but the building is done using msbuild of the Visual Studio solution.

For running tests, the fetching step also downloads the sample and netlib data. However, the step to extract the netlib mps.gz files to just mps is not part of fetch--AFAIK, extracting is normally doing as part of coinbrew _build_ of the project as a whole, which I won't do because I'm building using msbuild.

I tried to `coinbrew build Data/netlib` but [is not supported, apparently](https://github.com/jhmgoossens/Clp/actions/runs/4454682444/jobs/7824031901#step:8:221).
I can manually add a step to extract the mps.gz. 
Or is there a way to use coinbrew to 'build' (extract) the netlib and miplib datasets? 
Any suggestions? For now, I'll just skip the netlib and miplib tests. 
Many thanks for any help.

**Comment by tkralphs** [comment, marked as answer]
I can certainly add the option to "build" data projects to coinbrew. Can you open an issue for that? But for a single project with no dependencies, all it's going to do is call `configure; make; make install`, so you could certainly just do that instead of calling coinbrew. You may need to add `--build=x86_64-w64-mingw32` and if you want the files to be unzipped, then you need to configure with `--disable-zlib`. Use `--prefix` to specify where you want the files to be installed.

**Comment by jhmgoossens** [comment]
Thank you for considering. I created issue #82.
Until then I have simply added the hardcoded shortcut 
` if [ -d "./Data/Netlib/" ]; then gunzip ./Data/Netlib/*.gz; fi` 
in the [windows-ci.yml of Clp](https://github.com/coin-or/Clp/blob/558741ef6485c88e1ee64bb74c2768ce0ecf3ae6/.github/workflows/windows-ci.yml#L70). And honestly, since I don't believe the netlib and miplib3 will change significantly, this will probably work well for a long time (....) Thus, I understand that addressing #82 is very low priority.
Thanks again for your reply.
